src/legacy/core_plugins/metrics/server/plugin.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { InternalCoreSetup } from 'kibana/server';
```

Comment:
We should probably be importing `CoreSetup` here (and elsewhere) instead of `InternalCoreSetup` (afaik... I don't really know the difference but `InternalCoreSetup` is marked `@internal` in the core code, so I assume it isn't safe for consumption by plugins)

Reply:
@joshdover Could I ask you why it was implemented this way at `x-pack/legacy/plugins/apm/index.ts`?

Reply:
This is just temporary. As of now, `http` hasn't been exposed on CoreSetup yet because the final interface is still in progress. You can use `InternalCoreSetup` in the meantime to get access to this, just know that the interface is going to change between now and 7.4 FF.

Reply:
TIL. Thanks for clarifying @joshdover!